src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -22,11 +22,13 @@ namespace gcs {
 
 GcsNodeManager::NodeFailureDetector::NodeFailureDetector(
     boost::asio::io_service &io_service, gcs::NodeInfoAccessor &node_info_accessor,
+    std::shared_ptr<gcs::GcsPubSub> &gcs_pub_sub,
```

Comment:
Better to remove the reference, like this: `std::shared_ptr<gcs::GcsPubSub> gcs_pub_sub`, and  initialize `gcs_pub_sub_` like: `gcs_pub_sub_(std::move(gcs_pub_sub))`

Reply:
fixed